flaxandteal-arches/crimson
# Move import single csv for doorstep to a separate application

### What does this PR do?

Moves the ETL module and logic for processing to this new Arches Application: [arches_doorstep](https://github.com/flaxandteal/arches_doorstep)

For now, despite the discussion about calling remotely, I thought we could see if this works for @kanika-miglani for now, since it at least separates out the dependencies from the core project - we can define general purpose processors here: [arches_doorstep.inbuilt_processors](https://github.com/flaxandteal/arches_doorstep/tree/master/arches_doorstep/inbuilt_processors) and put their dependencies into the [pyproject.toml](https://github.com/flaxandteal/arches_doorstep/blob/11b156f7c7794da19920206299599382cbfe83ae/pyproject.toml#L45).

We may want to remove the Spacy NLP model, as it is 0.5G, but maybe that is useful out-of-the-box for now.

I have made a change to the `docker` submodule, which will mount `arches_doorstep` if available from the directory above, so you probably want to pull it (and clone arches_doorstep locally to the neighbouring directory for testing).

The steps to get the Arches Application working are in its README, although it involves execing onto the Docker container to run the `manage.py` commands. Doing those should load the ETL module at the same time as registering the Application, etc.

The chosen processors (which will probably be one from @kanika-miglani or maybe a couple, in the end) are [in the settings.py](https://github.com/flaxandteal-arches/crimson/blob/269a5bdaaf131d36af70ec583f63625fc7a7fb4c/crimson/settings.py#L424) Note that any new "processor" **must** be listed in [Arches Doorstep's config](https://github.com/flaxandteal/arches_doorstep/blob/11b156f7c7794da19920206299599382cbfe83ae/arches_doorstep/apps.py#L9) as a safety mechanism.

**Warning**: I have not touched the output JSON -- the array you are looking for is probably `results.table[0].warnings` array:

![Screenshot From 2024-12-07 22-29-17](https://github.com/user-attachments/assets/5b78de0c-2e89-4ea3-bb8c-732490e15436)

For now, it is likely that only the `Cell` type errors/warnings/info are relevant, which will have a `row` and `column` index in the JSON object in the warnings (/info/errors) array.

@kanika-miglani if you try installing `ltldoorstep` on Jupyter and starting a console (say), then you can call doorstep using `ltldoorstep process /home/philtweir/Cód/Oscailte/soirbhiochas/wikt-irish-prefixes/countries-ga-url.csv arches_doorstep/inbuilt_processors/pii_checker.py -e dask.threaded` or similar, so you can test without having to run the full Arches system. Happy to pick up in chat.

### How to test

After cloning this, try rebuilding the docker image (make sure the submodule to up-to-date). The instructions for registering the application are on the Arches Doorstep README, but you may need to unregister your local ETL module - `python manage.py etl_module unregister -n "Import Single CSV with processing"` - first.

Credit @chiatt for the Application template. Note that this is still pretty rough, and I have done the bare minimum to update Doorstep. However, it _does_ work for HTTP with Python 3.12, so if we need to swap it over, that should be pretty quick.
